31_HLSLPathTracer/main.cpp

```diff
@@ -420,13 +441,27 @@ class HLSLComputePathtracer final : public examples::SimpleWindowedApplication,
 					return shader;
 				};
 
+				auto getComputePipelineCreationParams = [](IGPUShader* shader, IGPUPipelineLayout* pipelineLayout) -> IGPUComputePipeline::SCreationParams
+				{
+					IGPUComputePipeline::SCreationParams params = {};
+					params.layout = pipelineLayout;
+					params.shader.shader = shader;
+					params.shader.entryPoint = "main";
+					params.shader.entries = nullptr;
+					params.shader.requireFullSubgroups = true;
+					params.shader.requiredSubgroupSize = static_cast<IGPUShader::SSpecInfo::SUBGROUP_SIZE>(5);
```

Comment:
this is bad and breaks on other GPUs than NV and RDNA, I know its @keptsecret's "fault" but lets fix, make the subgroup size the log2 of the minimum the device reports (for best divergence handling)

Reply:
> > this is bad and breaks on other GPUs than NV and RDNA, I know its @keptsecret's "fault" but lets fix, make the subgroup size the log2 of the minimum the device reports (for best divergence handling)
> 
> how do i query this?

its in physical device limits, copy it from FFT example
https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/blob/e5d5ae2ca9137a6966d00aa039f3e6dae7c23fb9/28_FFTBloom/main.cpp#L732

but make it the minimum device supports instead